src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs

```diff
@@ -99,13 +99,23 @@ private void OnNodeViewUnloaded(object sender, RoutedEventArgs e)
         /// <param name="eventArgs"></param>
         private void OnSizeChanged(object sender, EventArgs eventArgs)
         {
-            if (ViewModel != null)
+            if (ViewModel == null) return;
+
+            if (ViewModel.IsGhostedNode)
             {
-                var size = new double[] { ActualWidth, nodeBorder.ActualHeight };
-                if (ViewModel.SetModelSizeCommand.CanExecute(size))
-                {
-                    ViewModel.SetModelSizeCommand.Execute(size);
-                }
+                // dragged node image size should correspond to its source node
+                var width = ViewModel.NodeModel.Width;
```

Comment:
I'm not sure I want to use `Width` and `Height` from the `NodeModel` because future refactoring will, most likely, remove these properties from the model.

Can we use the fact that we are passing in information about the way that the `NodeViewModel` is being used, to pass an optional `PreferredSize` parameter? Is it possible to get the size information from the views that are being copied, and push it into these view models using that parameter?

This will also remove the ambiguity of the `IsGhostedNode` property. The result of being "ghosted" is that the view corresponding to this viewmodel needs to be sized in a certain way.